src/lib.rs

```diff
@@ -423,6 +423,56 @@ impl Build {
                 configure.arg("-D__STDC_NO_ATOMICS__");
             }
 
+            if target.contains("wasi") {
+                configure.args([
+                    // Termios isn't available whatsoever on WASM/WASI so we disable that
+                    // fatal error: 'termios.h' file not found
+                    "no-ui-console",
+                    // WASI doesn't support UNIX sockets so we preemptively disable it
+                    "no-sock",
+                    // WASI doesn't have a concept of syslog, so we disable it
+                    // fatal error: 'syslog.h' file not found
+                    "-DNO_SYSLOG",
+                    // wasm lacks signal support; to enable minimal signal emulation, compile with
+                    // -D_WASI_EMULATED_SIGNAL and link with -lwasi-emulated-signal
```

Comment:
If these `-l` flags are required with these compilation flags I think that the build script here and/or the build results will need to print out appropriate cargo directives to indicate that these extra libraries should be linked.

Reply:
I'll add those to be emitted for the sake of completeness. But from my experience playing with WASI, usually only the compilation directives are really required. (I get the "unused linker-arg" warning saying that those flags are ignored altogether most of the time)

Reply:
Update: Those flags are now emitted but it seems this feature requires `-Zextra-link-arg` on `cargo`. 

I get the following warnings now: 
```
warning: cargo:rustc-link-arg requires -Zextra-link-arg flag
```

And again, the build & test run works fine without those link args. Only the compilation directives are a hard requirement.

Reply:
Seems to me that it would be cleaner to omit the unnecessary linker flags and avoid the warnings.

Reply:
I'd favor this approach too; Also conceptually linker flags are usually the end consumer (= binary-producing)'s responsibility, so I don't think we need to bother with linkage in general.

Reply:
If a warning is printed out about the library not being needed then presumably it's not needed? In that case it should be fine to just remove the comment here about the need for the `-l` on the library.

Reply:
I did it with `println!("cargo:rustc-link-lib=flag-without-dash-L")`; Not sure what it does in practice but no warnings are printed neither from clang, rustc or lld whatsoever. I guess this is the correct way of doing things?

Reply:
I believe what's implemented in this PR right now, which is to do `println!("cargo:rustc-link-lib=wasi-emulated-signal");`, is what needs to be done. Does that print warnings or is there a raeson to not do that?

If the prints below stay then the comments here about libraries to link can probably be removed or just indicate that the corresponding library is linked below.

Reply:
No warnings, and indeed that's what I'm doing. I'll update the comments to reflect that.